### PR TITLE
Allow click prompt for SAM 2 

### PR DESCRIPTION
**Pull request by MichaelBuessemeyer**
### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- use quick-select in ai mode with click and bounding box

### Issues:
- follow-up for #7990 

------
(Please delete unneeded items, merge only when none are left open)
- [x] Updated [changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [ ] Updated [migration guide](../blob/master/MIGRATIONS.unreleased.md#unreleased) if applicable
- [x] Updated [documentation](../blob/master/docs) if applicable
- [ ] Adapted [wk-libs python client](https://github.com/scalableminds/webknossos-libs/tree/master/webknossos/webknossos/client) if relevant API parts change
- [ ] Removed dev-only changes like prints and application.conf edits
- [ ] Considered [common edge cases](../blob/master/.github/common_edge_cases.md)
- [ ] Needs datastore update after deployment

### CODE REVIEW

File: .circleci/not-on-master.sh [hunk 1]
```diff
@@ -1,8 +1,8 @@
...
 
-if [ "${CIRCLE_BRANCH}" == "master" ]; then
+# if [ "${CIRCLE_BRANCH}" == "master" ]; then
   echo "Skipping this step on master..."
-else
-  exec "$@"
-fi
+# else
+#   exec "$@"
+# fi
```
Comment (MichaelBuessemeyer) [inline .circleci/not-on-master.sh:8]: undo

File: conf/application.conf [hunk 2]
```diff
@@ -163,7 +163,7 @@ features {
   # The only valid item value is currently "ConnectomeView":
   optInTabs = []
   openIdConnectEnabled = false
-  segmentAnythingEnabled = false
+  segmentAnythingEnabled = true
```
Comment (MichaelBuessemeyer) [inline conf/application.conf:166]: undo

File: frontend/javascripts/oxalis/controller/combinations/tool_controls.ts [hunk 3]
```diff
@@ -726,6 +728,19 @@ export class QuickSelectTool {
 
         quickSelectGeometry.setCoordinates(startPos, currentPos);
       },
+      leftClick: (pos: Point2, _plane: OrthoView, _event: MouseEvent, _isTouch: boolean) => {
+        const state = Store.getState();
+        const startPos = V3.floor(calculateGlobalPos(state, pos));
```
Comment (MichaelBuessemeyer) [inline frontend/javascripts/oxalis/controller/combinations/tool_controls.ts:733]: This is a local variable correct? not the `startPos` part of the parent method `getPlaneMouseControls`. Thus, I'd like to improve the naming here as the clicked position is not a "startPos" but more something like "clickedPos" or "clickedPosGlobal" or "globalclickedPos"

```suggestion
        const clickedPos = V3.floor(calculateGlobalPos(state, pos));
```

File: frontend/javascripts/oxalis/model/sagas/quick_select_ml_saga.ts [hunk 4]
```diff
@@ -76,8 +77,19 @@ function* getMask(
...
+          pointX: userBoxRelativeToMaskInMag.getMinUV(activeViewport)[0],
+          pointY: userBoxRelativeToMaskInMag.getMinUV(activeViewport)[1],
+        }
+      : {
+          type: "BOUNDING_BOX",
+          selectionTopLeftX: userBoxRelativeToMaskInMag.getMinUV(activeViewport)[0],
+          selectionTopLeftY: userBoxRelativeToMaskInMag.getMinUV(activeViewport)[1],
+          selectionBottomRightX: userBoxRelativeToMaskInMag.getMaxUV(activeViewport)[0],
+          selectionBottomRightY: userBoxRelativeToMaskInMag.getMaxUV(activeViewport)[1],
+        },
```
Comment (MichaelBuessemeyer) [inline frontend/javascripts/oxalis/model/sagas/quick_select_ml_saga.ts:92]: Hmm maybe "extract" the calls to `userBoxRelativeToMaskInMax/Min` to have less duplication?

Kinda like:
```
const topLeft = userBoxRelativeToMaskInMag.getMinUV(activeViewport);
const bottomRight = userBoxRelativeToMaskInMag.getMaxUV(activeViewport);
```

And then reuse them in the yield call
```suggestion
    usePointPrompt
      ? {
          type: "POINT",
          pointX: topLeft[0],
          pointY: topLeft[1],
        }
      : {
          type: "BOUNDING_BOX",
          selectionTopLeftX: topLeft[0],
          selectionTopLeftY: topLeft[1],
          selectionBottomRightX: bottomRight[0],
          selectionBottomRightY: bottomRight[1],
        },
```

**Review by MichaelBuessemeyer — COMMENTED** [review]
Thanks for implementing the frontend part so fast @philippotto. 

Please find my potential improvements below

**Comment by philippotto** [comment]
@MichaelBuessemeyer could you have a look at my latest commits? I think, afterwards the PR might be good to merge.

**Review by philippotto — APPROVED** [review]

**Pull request merged**